Modules/_cursesmodule.c

```diff
@@ -211,6 +211,71 @@ static int curses_start_color_called = FALSE;
 
 static const char *curses_screen_encoding = NULL;
 
+/* Error type */
+
+#define CURSES_ERROR_FORMAT                 "%s() returned ERR"
+#define CURSES_ERROR_VERBOSE_FORMAT         "%s() (called by %s()) returned ERR"
+#define CURSES_ERROR_NULL_FORMAT            "%s() returned NULL"
+#define CURSES_ERROR_NULL_VERBOSE_FORMAT    "%s() (called by %s()) returned NULL"
+#define CURSES_ERROR_MUST_CALL_FORMAT       "must call %s() first"
+
+/* Utility Error Procedures */
+
+/*
+ * Format a curses error.
+ *
+ * A NULL 'python_funcname' falls back to 'curses_funcname' and vice-versa.
+ * If both names are NULL, the error message is 'catchall_ERR'.
+ */
+static void
+_PyCursesSetError(cursesmodule_state *state,
+                  const char *python_funcname,
+                  const char *curses_funcname)
+{
+    assert(!PyErr_Occurred());
+    if (python_funcname == NULL && curses_funcname == NULL) {
+        PyErr_SetString(state->error, catchall_ERR);
+    }
+    else if (python_funcname == NULL) {
+        PyErr_Format(state->error, CURSES_ERROR_FORMAT, curses_funcname);
+    }
+    else if (curses_funcname == NULL) {
+        PyErr_Format(state->error, CURSES_ERROR_FORMAT, python_funcname);
+    }
+    else {
+        PyErr_Format(state->error, CURSES_ERROR_VERBOSE_FORMAT,
+                     curses_funcname, python_funcname);
+    }
+}
+
+static void
+PyCursesSetError_ForWin_From(PyCursesWindowObject *win,
```

Comment:
Ideally, don't use `Py`, the prefix for public API, for new static functions.

Reply:
Maybe I should just have a `curses_set_error_*` prefix? I don't remember whether I unfortunately introduced this naming in this or in a previous PR :(